core/state/statedb.go

```diff
@@ -425,6 +507,18 @@ func (s *StateDB) MergeSlotDB(slotDb *StateDB, slotReceipt *types.Receipt, txInd
 			}
 			s.snapStorage[k] = temp
 		}
+		for k := range slotDb.snapDestructs {
+			obj, _ := s.loadStateObjectFromStateDB(k)
+			if !obj.suicided && !obj.empty() {
+				// There could be a race condition for parallel transaction execution
+				// One add balance 0 to an empty address, it will delete it(delete empty is enabled,).
+				// While another concurrent transaction could add a none-zero balance to it, make it not empty
+				// Right now, we mark a read operation to avoid this race condition
+				// We hope to remove this read record, since it could reduce conflict rate.
+				log.Warn("MergeSlotDB state object to be deleted is not empty, ", "addr", k)
+			}
+			s.snapDestructs[k] = struct{}{}
```

Comment:
do we also need to add these:

```
				delete(s.snapAccounts, obj.address)       // Clear out any previously updated account data (may be recreated via a ressurrect)
				delete(s.snapStorage, obj.address)        // Clear out any previously updated storage data (may be recreated via a ressurrect)
	
```

Reply:
As explained above, only remove the address from snapAccounts&snapStorage when object is deleted.
"snapDestructs" could be caused by createObject() to destroy previous object from trie on block commit.